schematics/types/base.py

```diff
@@ -791,6 +787,25 @@ def _mock(self, context=None):
             microsecond=random.randrange(1000000),
         )
 
+        # We return a timezone-aware datetime in two cases:
+        # 1. Half the time when timezone-aware and timezone-naive
+        #    datetime are allowed.
+        # 2. Always when timezone-aware datetime are required.
+        if ((self.tzd == 'allow' and random.randrange(2)) or
+                (self.tzd == 'require')):
+            # We use only hours for offset for simplicity.
+            hours = random.randrange(MIMIMUM_TZ_HOUR_OFFSET,
+                                     MAXIMUM_TZ_HOUR_OFFSET + 1)
+            mocked = mocked.replace(tzinfo=self.offset_timezone(hours=hours))
+
+        elif self.tzd == 'utc':
+            # Always return utc.
+            mocked = mocked.replace(tzinfo=self.UTC)
+
+        # If tzd == 'reject', we return a timezone-naive datetime.
+
+        return mocked
+
```

Comment:
Looks correct but doesn't take `convert_tz` or `drop_tzinfo` into account. Would you mind adding those?
* When `convert_tz` is true, the datetime can only be naive or UTC.
* When `drop_tzinfo` is true, the datetime can only be naive.